decaf377-ka/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+use ark_ff::UniformRand;
+
+use decaf377;
+use rand_core::{CryptoRng, RngCore};
+
+/// A `SharedSecret` derived at the end of the key agreement protocol.
+#[derive(Debug, PartialEq)]
+pub struct SharedSecret(pub(crate) decaf377::Element);
+
+impl SharedSecret {
+    pub fn derive(other_public_key: &EphemeralPublicKey, secret_key: EphemeralSecretKey) -> Self {
+        Self(other_public_key.0 * secret_key.0)
+    }
+}
+
+/// An `EphemeralSecretKey` is used once and consumed when forming a `SharedSecret`.
+pub struct EphemeralSecretKey(pub(crate) decaf377::Fr);
```

Comment:
Non-blocking for this PR, but we should think about how the incoming viewing key is supposed to interact with this API -- we probably also want to have a non-ephemeral secret key type, and have the IVK be an instance of that type.